src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/MSBuildProjectManager.cs

```diff
@@ -54,18 +52,18 @@ internal class MSBuildProjectManager : IMSBuildEventSink, IOmniSharpProjectSnaps
             _projectSnapshotManagerDispatcher = projectSnapshotManagerDispatcher;
         }
 
+        public OmniSharpProjectSnapshotManagerBase ProjectManager => _projectManager ?? throw new InvalidOperationException($"{nameof(ProjectManager)} called before {nameof(Initialize)}");
+
         public void Initialize(OmniSharpProjectSnapshotManagerBase projectManager!!)
         {
             _projectManager = projectManager;
         }
 
-#pragma warning disable VSTHRD100 // Avoid async void methods
-        public async void ProjectLoaded(ProjectLoadedEventArgs args)
-#pragma warning restore VSTHRD100 // Avoid async void methods
+        public void ProjectLoaded(ProjectLoadedEventArgs args)
         {
             try
             {
-                await ProjectLoadedAsync(args);
+                _ = ProjectLoadedAsync(args);
```

Comment:
The exception handling needs to be moved into ProjectLoadedAsync, its doing nothing here.